feat: add instructional hint banner after closing welcome tour

Inuth0603 opened the pull request:
Fixes #4943

Overview This PR addresses the "guidance gap" identified in Issue #4943. Currently, when a first-time user closes the "Welcome Tour," they are left with a cluttered workspace and no immediate instruction on how to begin.

Changes

Modified js/widgets/help.js to add an onclose logic listener to the HelpWidget.

Implemented a non-intrusive, localized hint banner that appears only after the initial "Take a Tour" page is closed.

The banner provides a clear call-to-action: "Start by dragging a block from the left panel and connect it to the Start block."

Added automatic fade-out logic (5 seconds) and DOM cleanup to keep the workspace tidy.

Added the necessary string to locales/en.json to support the localization system.

Testing Performed

UX Verification: Confirmed on localhost that the banner only triggers for the introductory tour (index === 0) and not for general block help.

Localization: Swapped languages on localhost and verified that the banner correctly displays translated strings (e.g., tested with Spanish).

Regression: Verified that this UI-only change does not affect core program logic or existing block functionality.
<img width="1920" height="1080" alt="Screenshot 2025-12-30 120635" src="https://github.com/user-attachments/assets/356393c7-c252-45a0-bfeb-12aa8da8f833" />

walterbender commented:
I like the general direction this is heading. Let's talk with @pikurasa about what might be most useful as a help message.

A couple of things:

(1) we are in string freeze right now, so I will not merge this until after the release planned for this week -- it is not ready yet in any case;
(2) when editing i18n strings, the changes should be made the PO files, not the JSON files, as the JSON files are generated from the PO files. If you edit directly in the JSON files, at some point those changes will be lost.

Finally, you made unrelated changes to the JSON files which belong in a separate PR.

Inuth0603 commented:
Hi @walterbender and @pikurasa,

Just a gentle follow-up now that tests are passing and the release/string freeze is past.
I’m happy to adjust the wording or behavior of the hint banner based on feedback.

Thanks for taking a look when you have time!

walterbender commented:
You changes to help.js are spot on. But you have two unrelated changes that need to be removed from this PR.

Inuth0603 commented:
Hi @walterbender,

Thank you for the feedback and patience! I’ve updated the PR with all the requested changes:

1. **Replaced** the custom DOM element with the standard message mechanism (`this.activity.textMsg()`) for the hint functionality.
2. **Reverted** `po/en.po` entirely to its upstream state. The instruction text is no longer added manually and will be picked up automatically when the master POT file regenerates.
3. **Cleaned up history**: rebased onto `upstream/master` and squashed everything into a single clean commit.
4. **Formatting**: CI was failing due to `eslint --max-warnings=0` catching some pre-existing loose equality (`==`) in `js/widgets/help.js`. I’ve updated those to strict equality (`===`) so the build passes cleanly.

Ready for another look whenever you have time
